src/BenchmarksServer/Startup.cs

```diff
@@ -4082,6 +4087,107 @@ private static async Task UseMonoRuntimeAsync(string runtimeVersion, string outp
+                }
+
+                Log.WriteLine($"Patching local dotnet with mono runtime and extracting llvm");
+
+                var packageName = "Microsoft.NETCore.App.Runtime.Mono.LLVM.AOT.linux-x64".ToLowerInvariant();
+                var runtimePath = Path.Combine(_rootTempDir, "RuntimePackages", $"{packageName}.{runtimeVersion}.nupkg");
+
+                var strCmdGetVer = "./dotnet --list-runtimes | grep -i \"Microsoft.NETCore.App\"";
+                var resultGetVer = runBashCmd(strCmdGetVer, Path.GetDirectoryName(dotnetMonoPath));
+                var MicrosoftNETCoreAppPackageVersion = resultGetVer.Split(' ')[1];
+
+                var llvmExtractDir = Path.Combine(Path.GetDirectoryName(runtimePath), "mono-llvm");
+                if (Directory.Exists(llvmExtractDir))
+                {
+                    Directory.Delete(llvmExtractDir);
+                }
+                Directory.CreateDirectory(llvmExtractDir);
+
+                using (var archive = ZipFile.OpenRead(runtimePath))
+                {
+                    var systemCoreLib = archive.GetEntry("runtimes/linux-x64/native/System.Private.CoreLib.dll");
+                    systemCoreLib.ExtractToFile(Path.Combine(Path.GetDirectoryName(dotnetMonoPath), "shared", "Microsoft.NETCore.App", MicrosoftNETCoreAppPackageVersion, "System.Private.CoreLib.dll"), true);
+
+                    var libcoreclr = archive.GetEntry("runtimes/linux-x64/native/libcoreclr.so");
+                    libcoreclr.ExtractToFile(Path.Combine(Path.GetDirectoryName(dotnetMonoPath), "shared", "Microsoft.NETCore.App", MicrosoftNETCoreAppPackageVersion, "libcoreclr.so"), true);
+
+                    var llcExe = archive.GetEntry("runtimes/linux-x64/native/llc");
+                    llcExe.ExtractToFile(Path.Combine(llvmExtractDir, "llc"), true);
+
+                    var optExe = archive.GetEntry("runtimes/linux-x64/native/opt");
+                    optExe.ExtractToFile(Path.Combine(llvmExtractDir, "opt"), true);
+                }
+                var strChmod = "chmod +x opt llc";
+                var resultChmod = runBashCmd(strChmod, llvmExtractDir);
+
+                Log.WriteLine("Pre-compile assemblies inside publish folder");
+
+                var strCmdPreCompile = $@"for assembly in {outputFolder}/*.dll; do
+                                              PATH={llvmExtractDir}:${{PATH}} MONO_ENV_OPTIONS=--aot=llvm,mcpu=native ./dotnet $assembly;
+                                           done";
+                var resultPreCompile = runBashCmd(strCmdPreCompile, Path.GetDirectoryName(dotnetMonoPath));
+                Log.WriteLine("=========================");
+                Log.WriteLine(resultPreCompile);
+
+            }
+            catch (Exception e)
+            {
+                Log.WriteLine("ERROR: Failed to AOT for mono. " + e.ToString());
+                throw;
+            }
+        }
+
+	private static string runBashCmd(string cmd, string wd)
```

Comment:
There is already a `ProcessUtil.RunAsync` that is used across this file to run commands. Search for `cgroup`, or `GetSwapBytes`